backend/app/services/adapters/workspace_archive.py

```
@@ -0,0 +1,429 @@
+    @property
+    def s3_client(self):
+        """Lazy initialization of S3 client."""
+        if self._s3_client is None:
+            self._s3_client = get_workspace_s3_client()
+        return self._s3_client
+
+    def archive_workspace(
+        self,
+        db: Session,
+        task_id: int,
+        executor_name: str,
```

Ruff (0.14.14)

[warning] Unused method argument: `executor_name`

(ARG002)